File: autos-stock-client/src/app/models/DepenseDashboardVm.ts
import {PageVm} from "./PageVm";
import {DepenseDto} from "./DepenseDto";
import {DepenseMonthlyTotalDto} from "./DepenseMonthlyTotalDto";

export interface DepenseDashboardVm {
  total: number;
  totalPeriode?: number | null;
  page: PageVm<DepenseDto>;
  monthly?: DepenseMonthlyTotalDto[];
}

File: Autos-stock-manager-services/src/main/java/org/autostock/dtos/DepenseDashboardVm.java
package org.autostock.dtos;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
public class DepenseDashboardVm {
    private BigDecimal total;                // total global voiture
    private BigDecimal totalPeriode;         // total période si fournie
    private PageVm<DepenseDto> page;         // liste paginée
    private List<DepenseMonthlyTotalDto> monthly; // graph mensuel
    private List<DepenseMonthlyByCategorieDto> monthlyByCategorie; // graph par catégorie

}
